### PR TITLE
Two new alternatives to measure_with_pause task

### PR DESCRIPTION
**Pull request by HHeda**
IterateTask resumes the program and inputs True into input stream "iterate" to iterate a while loop in the program. Waits for the program to be paused again.
FinishTask resumes the program, uses the same input stream to input False, break the loop to finish the program and fetch all the data

ResumeTask and ResumeAndGetData do the same without the input stream, so the while loop should be a for loop with the right number of repetitions

### CODE REVIEW

**Comment by rassouly** [comment]
There is no `input_stream` "iterate" used. Is that normal?
EDIT: If you are worried about compatibility with older OPXs, wrap the call in a try-except that logs out a warning

**Comment by rassouly** [comment]
Otherwise, looks good!

File: exopy_qm/instruments/drivers/QuantumMachine.py [hunk 1]
```diff
@@ -112,6 +116,24 @@ def is_paused(self):
     def resume(self):
         self.job.resume()
 
+    def wait_for_pause(self):
+        """waits for the program to be paused
+        """
+        while not self.is_paused():
+            time.sleep(0.01)
+
```
Comment (HHeda) [inline exopy_qm/instruments/drivers/QuantumMachine.py:124]: The use of the input streams are here

File: exopy_qm/instruments/drivers/QuantumMachine.py [hunk 2]
```diff
@@ -112,6 +116,24 @@ def is_paused(self):
...
+        """
+        self.job.resume()
+        self.job.insert_input_stream('iterate', True)
+
+    def finish(self):
+        """Finishes the program by resuming it and feeding False to the input sting 'iterate'
+          """
+        self.job.resume()
+        self.job.insert_input_stream('iterate', False)
+
```
Comment (HHeda) [inline exopy_qm/instruments/drivers/QuantumMachine.py:136]: The uses of input streams are in the driver, just above

Comment (rassouly) [reply exopy_qm/instruments/drivers/QuantumMachine.py:136]: Oups, didn't see that line

**Comment by rassouly** [comment]
Merged! Thank you

**Pull request merged**